mapbox-lint/src/main/java/com/mapbox/maps/lint/LifecycleMethodDetector.kt

```diff
@@ -0,0 +1,46 @@
+package com.mapbox.maps.lint
+
+import com.android.tools.lint.detector.api.*
+import com.intellij.psi.PsiMethod
+import org.jetbrains.uast.UCallExpression
+
+/**
+ * Lint detector for lifecycle events
+ */
+class LifecycleMethodDetector : Detector(), Detector.UastScanner {
+
+  override fun getApplicableMethodNames()= listOf(ON_START, ON_STOP, ON_DESTROY, ON_LOW_MEMORY)
+
+  override fun visitMethodCall(context: JavaContext, node: UCallExpression, method: PsiMethod) {
```

Comment:
nit: formatting, have we enabled ktlint for this module?

Reply:
Guess no